### ISSUE TITLE
Duplicate snapshots are created on each run, previous snapshots ignored

### ISSUE THREAD

**Issue by mattfysh** [oven-sh/bun#14029]
### What version of Bun is running?

1.1.28+cf4e9cb69

### What platform is your computer?

Darwin 23.6.0 arm64 arm

### What steps can reproduce the bug?

```js
import { test, expect } from 'bun:test'

test('foo', () => {
  expect('bar').toMatchSnapshot()
})
```

then repeatedly run `bun test` and watch the snapshot file fill up with duplicates
you can also change 'bar' to something else and the test should fail but it doesn't.

**Comment by wincent** [comment]
Also reproduces for me in 1.1.28, and looks to be a new regression (not in 1.1.27). Taking [a snapshot file from one of my projects](https://github.com/wincent/masochist/blob/next/packages/lexer/src/__tests__/__snapshots__/union.test.ts.snap) and running `bun test` 10 times, I see the following (ie. one duplicate added per `bun test` run):

```
$ for X in $(seq 1 10); do bun test; done 
$ grep exports packages/lexer/src/__tests__/__snapshots__/union.test.ts.snap
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
exports[`union() prefers greedy matches 1`] = `
```

This regression seems much more general and unrelated to the snapshot duplication I previously reported here:

- https://github.com/oven-sh/bun/issues/4722

**Comment by rhuanbarreto** [comment]
I can confirm it's a regression introduced in Bun v1.1.28. Downgrading to v1.1.27 fix it.

**Comment by gabrieljablonski** [comment]
Seems to have been fixed since 1.1.29 (we're on 1.1.30 at the time of writing).

Run `bun upgrade` and be happy.
